x-pack/plugins/alerts/server/lib/license_state.ts

```diff
@@ -37,6 +47,47 @@ export class LicenseState {
     return this.licenseInformation;
   }
 
+  public setNotifyUsage(notifyUsage: LicensingPluginStart['featureUsage']['notifyUsage']) {
+    this._notifyUsage = notifyUsage;
+  }
+
+  public isLicenseValidForAlertType(
+    alertTypeId: string,
+    alertTypeName: string,
+    minimumLicenseRequired: LicenseType,
+    { notifyUsage }: { notifyUsage: boolean } = { notifyUsage: false }
+  ): { isValid: true } | { isValid: false; reason: 'unavailable' | 'expired' | 'invalid' } {
+    if (notifyUsage) {
+      this.notifyUsage(alertTypeName, minimumLicenseRequired);
+    }
```

Comment:
I definitely wouldn't expect a method that starts with `is...` to have a side effect like this.
Should the notification be extracted from here and have it's own entry point?

Reply:
Renamed to getLicenseCheckForAlertType